internal/assets/token_approval.go

```diff
@@ -58,8 +59,9 @@ func (s *approveSender) setDefaults() {
 
 func (am *assetManager) NewApproval(approval *core.TokenApprovalInput) syncasync.Sender {
 	sender := &approveSender{
-		mgr:      am,
-		approval: approval,
+		mgr:              am,
+		approval:         approval,
+		idempotentSubmit: approval.IdempotencyKey != "",
```

Comment:
I would have to do more investigation to convince myself of that.. and if I did so, I'd probably want to rename `approval` to `approvalInput` (rather than the implying it's the actual approval object).

Reply:
So... if it's ok, think I'd like to leave it as is for this PR

Reply:
But yes, the thing we store is the input DTO from the user, so possibly it's named incorrectly.